scripts/intfutil

```diff
@@ -267,13 +298,42 @@ def appl_db_portchannel_status_get(appl_db, config_db, po_name, status_type, por
         return "N/A"
     return status 
 
+def appl_db_sub_intf_status_get(appl_db, config_db, front_panel_ports_list, portchannel_speed_dict, sub_intf_name, status_type):
+    sub_intf_sep_idx = sub_intf_name.find(VLAN_SUB_INTERFACE_SEPARATOR)
+    if sub_intf_sep_idx != -1:
+        parent_port_name = sub_intf_name[:sub_intf_sep_idx]
+        vlan_id = sub_intf_name[sub_intf_sep_idx + 1:]
+
+        full_intf_table_name = "INTF_TABLE" + ":" + sub_intf_name
+
+        if status_type == "vlan":
+            return vlan_id
+
+        if status_type == "admin_status":
+            status = appl_db.get(appl_db.APPL_DB, full_intf_table_name, status_type)
+            return status if status is not None else "N/A"
+
+        if status_type == "type":
+            return VLAN_SUB_INTERFACE_TYPE
+
+        if status_type == "mtu" or status_type == "speed":
+            if parent_port_name in front_panel_ports_list:
+                return appl_db_port_status_get(appl_db, parent_port_name, status_type)
+            elif parent_port_name in portchannel_speed_dict.keys():
+                return appl_db_portchannel_status_get(appl_db, config_db, parent_port_name, status_type, portchannel_speed_dict)
+            else:
+                return "N/A"
+
+    return "N/A"
+
 # ========================== interface-status logic ==========================
 
 header_stat = ['Interface', 'Lanes', 'Speed', 'MTU', 'Alias', 'Vlan', 'Oper', 'Admin', 'Type', 'Asym PFC']
+header_stat_sub_intf = ['Sub port interface', 'Speed', 'MTU', 'Vlan', 'Admin', 'Type']
```

Comment:
>Sub port interface [](start = 25, length = 18)

I am confused about the terminology `Sub port interface` vs `subinterfaces`

Reply:
It is interchangeable. In SAI standard, the RIF type is sub port. https://github.com/opencomputeproject/SAI/blob/master/doc/bridge/SAI_Proposal_Bridge_port_v0.9.docx That's why sub port interface is used here. Not sure about the standard term for sub port interface.